### PR TITLE
feat(FX-3614): Update visual design of artwork grid filter menu

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Feature**

This PR resolves [FX-3614]

### Acceptance Criteria
* Update the visual design to match the spec in [Figma](https://www.figma.com/file/zbEFZqjKgKquZwtU8U44j9/Alerts?node-id=3321%3A6866)
* Only the "Apply Filters" button should be displayed at the bottom (without “Create Alert“ button)
* Artwork grid filter menu should open at full height
* Top header on the filters screen (for example, Materials, Medium, Sort By) should match the spec in [Figma](https://www.figma.com/file/zG9cduJdOaTQAw886QcOwn/Filters?node-id=2737%3A11807)
* Updated design should be displayed on all screens, not just for artist artworks

#### iOS
https://user-images.githubusercontent.com/3513494/145037815-e3412871-36f7-4605-b7a7-b43bc2b66929.mp4

#### Android
https://user-images.githubusercontent.com/3513494/145037969-141e45ee-3295-4ede-b871-89d640d9770c.mp4

|| iOS | Android |
|-|-|-|
|Sort & Filter|Simulator Screen Shot - iPhone 8 - 2021-12-07 at 16 33 13|photo_2021-12-07 16 37 35|
|Sort & Filter with selected filters|Simulator Screen Shot - iPhone 8 - 2021-12-07 at 16 33 21|photo_2021-12-07 16 37 35 (1)|
|Medium filter|Simulator Screen Shot - iPhone 8 - 2021-12-07 at 16 35 19|photo_2021-12-07 16 37 34|
|Medium filter with selected values|Simulator Screen Shot - iPhone 8 - 2021-12-07 at 16 33 37|photo_2021-12-07 16 37 36|
|Material filter (with search box)|Simulator Screen Shot - iPhone 8 - 2021-12-07 at 16 35 58|photo_2021-12-07 16 37 32|

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Update visual design of artwork grid filter menu - dimatretyak

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[FX-3614]: https://artsyproduct.atlassian.net/browse/FX-3614?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Update visual design of artwork grid filter menu - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 74f6f423709b0502578c7e81e5e03c73ba448cc2
</p>

File: src/lib/Components/Gene/GeneArtworks.tests.tsx [hunk 1]
```diff
@@ -1,17 +1,10 @@
+import { fireEvent } from "@testing-library/react-native"
```
Comment (gkartalis) [inline src/lib/Components/Gene/GeneArtworks.tests.tsx:1]: **praise:** 🤩

**Review by gkartalis — APPROVED** [review]
🚀  Amazing job! Thanks for updating the tests and tidying everything up 🧹

File: src/lib/Components/ArtworkFilter/ArtworkFilter.tsx [hunk 2]
```diff
@@ -101,6 +101,7 @@ export const ArtworkFilterNavigator: React.FC<ArtworkFilterProps> = (props) => {
 
   const applyFiltersAction = ArtworksFiltersStore.useStoreActions((action) => action.applyFiltersAction)
   const resetFiltersAction = ArtworksFiltersStore.useStoreActions((action) => action.resetFiltersAction)
+  const isEnabledImprovedAlertsFlow = useFeatureFlag("AREnableImprovedAlertsFlow")
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/lib/Components/ArtworkFilter/ArtworkFilter.tsx:104]: > We want to do that everywhere that we show the artwork grid or not yet

I clarified this question with the designers and got the answer that the new design should be in all places (you can find answer in "question" section [here](https://artsyproduct.atlassian.net/browse/FX-3614))

[the remark this answers is not in the document]
Comment (gkartalis) [reply src/lib/Components/ArtworkFilter/ArtworkFilter.tsx:104]: Thanks Dima!

[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/lib/Components/ArtworkFilter/ArtworkFilter.tsx:104]: > we change all the filter modals inside the app right

Yep, you're right

**Review by gkartalis — APPROVED** [review]
🚀

**Pull request merged**